### PR TITLE
#1793 - implement reverse style

### PR DESCRIPTION
**Pull request by grissly-man**
Reverses the fg and bg colors, if defined. If only fg is defined, currently sets background to the foreground color, and returns foreground to the default style

### CODE REVIEW

**Comment by xPaw** [comment]
This PR is a bit screwed since you based it on the same branch as your previous PR which brings in incorrect commits here.

File: client/js/libs/handlebars/parse.js [hunk 1]
```diff
@@ -16,11 +16,20 @@ function createFragment(fragment) {
 	if (fragment.bold) {
 		classes.push("irc-bold");
 	}
-	if (fragment.textColor !== undefined) {
-		classes.push("irc-fg" + fragment.textColor);
-	}
-	if (fragment.bgColor !== undefined) {
-		classes.push("irc-bg" + fragment.bgColor);
+	if (fragment.reverse) {
```
Comment (xPaw) [inline client/js/libs/handlebars/parse.js]: I think it's better to implement these in `parseStyle` so the fragments can be optimized correctly.

File: client/js/libs/handlebars/parse.js [hunk 2]
```diff
@@ -39,10 +48,18 @@ function createFragment(fragment) {
 	const escapedText = Handlebars.Utils.escapeExpression(fragment.text);
 
 	if (fragment.hexColor) {
-		attributes += ` style="color:#${fragment.hexColor}`;
+		if (fragment.reverse) {
```
Comment (xPaw) [inline client/js/libs/handlebars/parse.js]: I don't think we need to reverse hex colors for this. http://modern.ircdocs.horse/formatting.html#reverse-color

File: client/views/windows/help.tpl [hunk 3]
```diff
@@ -37,6 +37,18 @@
 			</p>
 		</div>
 	</div>
+	
+	<div class="help-item">
+		<div class="subject">
+			<kbd class="key-all">Ctrl</kbd><kbd class="key-apple">⌘</kbd> + <kbd>R</kbd>
```
Comment (xPaw) [inline client/views/windows/help.tpl]: Ctrl+R is reload page which is useful, overall I don't think keybind is needed for this at all.

File: client/js/libs/handlebars/parse.js (hunk 1 above)
Comment (grissly-man) [reply client/js/libs/handlebars/parse.js]: Happy to do this, but if we implement these in `parseStyle` doesn't that render the `reverse` flag in the fragment moot? Namely all logic/testing can be done via the `textColor` and `bgColor` and `reverse` seems somewhat redundant

Comment (xPaw) [reply client/js/libs/handlebars/parse.js]: > doesn't that render the reverse flag in the fragment moot?

I believe so, yeah. Feel free to remove it.

File: test/client/js/libs/handlebars/parse.js [hunk 4]
```diff
@@ -243,6 +243,18 @@ describe("parse Handlebars helper", () => {
 		name: "monospace",
 		input: "\x11monospace",
 		expected: '<span class="irc-monospace">monospace</span>',
+	}, {
+		name: "reverse with foreground and background colors",
+		input: "\x0313,1fg and bg \x16and reversed",
```
Comment (xPaw) [inline test/client/js/libs/handlebars/parse.js]: I would add tests like `\x0313,01test\x16test\x16test`, `\x0313,01test\x16test\x01,13test` to see what kind of html it generates.

**Comment by grissly-man** [comment]
@xPaw So currently this is set to toggle reverse until a new color formatting is specified. At that point it will parse the new formatting as normal, and not as though it is reversed. Is this the behavior you want implemented?

Also, the keybinding and help document have been reverted

**Comment by xPaw** [comment]
I think that is expected, yeah.

Can you rebase correctly upon thelounge's master? Your PR currently has unnecessary commits in it.

**Comment by grissly-man** [comment]
@xPaw Yessir

**Comment by astorije** [comment]
Hey @grissly-man, we have free sticker packs for our contributors now!
If you're interested, go fill the form at https://goo.gl/forms/f5usqAEp5DWoeXC92 🙂

**Pull request merged**